scripts/operations_tests/operations_tests.py

```diff
@@ -172,6 +177,18 @@ def add_nan(self, image_stack, fraction=0.001):
         data.ravel()[nan_indices] = np.nan
         return image_stack
 
+    def add_flats_and_darks(self, params):
+        for frame_type in ['flat_before', 'dark_before', 'flat_after', 'dark_after']:
+            if frame_type in params:
+                filename_group = FilenameGroup.from_file(config_manager.load_sample)
+                related_filename_group = filename_group.find_related(getattr(FILE_TYPES, frame_type.upper()))
+                related_filename_group.find_all_files()
```

Comment:
I get `AttributeError: 'NoneType' object has no attribute 'find_all_files`, so looks like you need to check if `related_filename_group` is None (which will happen if it does not find that sort of related stack)

Reply:
The test dataset only has Flat_Before, Flat_After, Dark_Before